app/src/main/java/com/islaharper/dawnofandroid/presentation/screens/welcome/WelcomeViewModel.kt

```diff
@@ -0,0 +1,20 @@
+package com.islaharper.dawnofandroid.presentation.screens.welcome
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.islaharper.dawnofandroid.domain.useCases.saveOnboarding.SaveOnBoardingStateUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class WelcomeViewModel @Inject constructor(
+    private val saveOnBoardingStateUseCase: SaveOnBoardingStateUseCase,
+) : ViewModel() {
+    fun saveOnBoardingState(completed: Boolean) {
+        viewModelScope.launch(Dispatchers.IO) {
```

Comment:
Should the ViewModel be choosing the dispatcher (main safety)?

Reply:
Ah yes, this one slipped by, we discussed and resolved this in an earlier PR.